astroquery/esa_hubble/core.py

```diff
@@ -0,0 +1,374 @@
+            optional, default None
+            File name for the artifact
+            verbose : bool, optional, default 'False'
+            Flag to display information about the process
+
+            Returns
+            -------
+            None. It downloads the observation postcard indicated
+        """
+
+        retri_type = "RETRIEVAL_TYPE=POSTCARD"
+        obs_id = "OBSERVATION_ID=" + observation_id
+        cal_level = "CALIBRATION_LEVEL=" + calibration_level
+        res = "RESOLUTION=" + str(resolution)
+        link = "".join((self.data_url,
+                        retri_type,
+                        "&",
+                        obs_id,
+                        "&",
+                        cal_level,
+                        "&",
+                        res))
+        if filename is None:
+            filename = observation_id + ".tar"
+        print(link)
+        return self._handler.get_file(link, filename, verbose)
+
+    def cone_search(self, coordinates, radius=0.0, filename=None,
+                    output_format='votable', verbose=False):
+        coord = self._getCoordInput(coordinates, "coordinate")
+        radiusInGrades = float(radius/60)  # Converts to degrees
+
+        raHours, dec = commons.coord_to_radec(coord)
+        ra = raHours * 15.0  # Converts to degrees
+        payload = {"RESOURCE_CLASS": "OBSERVATION",
+                   "ADQLQUERY": "SELECT DISTINCT OBSERVATION,OBSERVATION.TYPE,"
+                   "TARGET.MOVING_TARGET"
+                   ",TARGET.TARGET_NAME,TARGET.TARGET_DESCRIPTION,PROPOSAL."
+                   "PROPOSAL_ID,PROPOSAL.PI_"
+                   "NAME,PROPOSAL.PROPOSAL_TITLE,INSTRUMENT.INSTRUMENT_NAME,"
+                   "PLANE.METADATA_PROVENANCE"
+                   ",PLANE.DATA_PRODUCT_TYPE,PLANE.SOFTWARE_VERSION,POSITION"
+                   ".RA,POSITION.DEC,POSITION."
+                   "GAL_LAT,POSITION.GAL_LON,POSITION.ECL_LAT,POSITION.ECL_LON"
+                   ",POSITION.FOV_SIZE,ENERGY."
+                   "WAVE_CENTRAL,ENERGY.WAVE_BANDWIDTH,ENERGY.WAVE_MAX,ENERGY"
+                   ".WAVE_MIN,ENERGY.FILTER FROM"
+                   " FIELD_NOT_USED  WHERE OBSERVATION.COLLECTION='HST'  AND  "
+                   "PLANE.MAIN_SCIENCE_PLANE="
+                   "'true'  AND  (OBSERVATION.TYPE='HST Composite' OR "
+                   "OBSERVATION.TYPE='HST Singleton')"
+                   "  AND  INTERSECTS(CIRCLE('ICRS'," + str(ra) + ""
+                   "," + str(dec) + "," + str(radiusInGrades) + ""
+                   "),POSITION)=1  AND  PLANE.MAIN_SCIENCE_PLANE='true' "
+                   "ORDER BY PROPOSAL.PROPOSAL_ID "
+                   "DESC",
+                   # "PAGE": "1",
+                   # "PAGE_SIZE": "50",
+                   "RETURN_TYPE": str(output_format)}
+        result = urllib.parse.urlencode(payload,
+                                        quote_via=urllib.parse.quote_plus)
```

Comment:
this should use `requests` instead.  `urlencode` is not needed, you can just do `requests.get(url, payload=dict)` or similar.  In astroquery, you instead do `self._request('GET', url, payload=dict)` or similar